src/Illuminate/Routing/ImplicitRouteBinding.php

```diff
@@ -68,4 +47,73 @@ protected static function getParameterName($name, $parameters)
             return $snakedName;
         }
     }
+
+    /**
+     * Create an implicit route binding instance.
+     *
+     * @param  \Illuminate\Container\Container  $container
+     */
+    public function __construct($container)
+    {
+        $this->container = $container;
+    }
```

Comment:
This just doesn't seem right to inject the container here as a new dependency.

Reply:
The container is a dependency of implicit route binding (it needs to resolve bound items from the container). Since this PR is moving from a static call to an instance call, it needs to be passed in.